apis/config/v1beta1/defaults.go

```diff
@@ -73,10 +67,21 @@ func SetDefaults_Configuration(cfg *Configuration) {
 	if len(cfg.Health.HealthProbeBindAddress) == 0 {
 		cfg.Health.HealthProbeBindAddress = DefaultHealthProbeBindAddress
 	}
-	if cfg.LeaderElection != nil && cfg.LeaderElection.LeaderElect != nil &&
-		*cfg.LeaderElection.LeaderElect && len(cfg.LeaderElection.ResourceName) == 0 {
+
+	if cfg.LeaderElection == nil {
+		cfg.LeaderElection = &configv1alpha1.LeaderElectionConfiguration{}
+	}
+	if len(cfg.LeaderElection.ResourceName) == 0 {
 		cfg.LeaderElection.ResourceName = DefaultLeaderElectionID
 	}
+	if len(cfg.LeaderElection.ResourceLock) == 0 {
+		// Default to Lease as component-base still defaults to endpoint resources
+		// until core components migrate to using Leases. See k/k #80289 for more details.
+		cfg.LeaderElection.ResourceLock = resourcelock.LeasesResourceLock
```

Comment:
Is this a change compared to what is happening on the main branch? I'm wondering if this deserves a release note explaining what are the user-facing differences when using defaults.

Reply:
No, it doesn't change the default value compared to main. It's necessary so we can delegate to `RecommendedDefaultLeaderElectionConfiguration`, but that still defaults to endpoints as resource lock, because some Kubernetes components haven't migrated yet, and maintain the current default value for Kueue.
This can be removed once kubernetes/kubernetes#80289 is fixed.